flash/vision/style_transfer/model.py

```diff
@@ -0,0 +1,227 @@
+
+    def forward(self, input):
+        input = self.preprocessor(input)
+        output = self.decoder(self.encoder(input))
+        return self.postprocessor(output)
+
+
+class StyleTransfer(Task):
+
+    def __init__(
+        self,
+        style_image: Union[str, torch.Tensor],
+        model: Optional[nn.Module] = None,
+        multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None,
+        content_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        style_loss: Optional[Union[ops.ComparisonOperator, ops.OperatorContainer]] = None,
+        optimizer: Union[Type[torch.optim.Optimizer], torch.optim.Optimizer] = torch.optim.Adam,
+        optimizer_kwargs: Optional[Dict[str, Any]] = None,
+        scheduler: Optional[Union[Type[_LRScheduler], str, _LRScheduler]] = None,
+        scheduler_kwargs: Optional[Dict[str, Any]] = None,
+        learning_rate: float = 1e-3,
+        serializer: Optional[Union[Serializer, Mapping[str, Serializer]]] = None,
+    ):
+        self.save_hyperparameters(ignore="style_image")
+
+        if isinstance(style_image, str):
+            style_image = read_image(style_image)
+
+        if model is None:
+            # TODO: import this from pystiche
+            model = Transformer()
+
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+
+        if content_loss is None:
+            content_loss = self.default_content_loss(multi_layer_encoder)
+
+        if style_loss is None:
+            style_loss = self.default_style_loss(multi_layer_encoder)
+
+        perceptual_loss = loss.PerceptualLoss(content_loss, style_loss)
+
+        super().__init__(
+            model=model,
+            loss_fn=perceptual_loss,
+            optimizer=optimizer,
+            optimizer_kwargs=optimizer_kwargs,
+            scheduler=scheduler,
+            scheduler_kwargs=scheduler_kwargs,
+            learning_rate=learning_rate,
+            serializer=serializer,
+        )
+
+        self.perceptual_loss = perceptual_loss
+        self.perceptual_loss.set_style_image(style_image)
+
+    def default_multi_layer_encoder(self) -> enc.MultiLayerEncoder:
+        return enc.vgg16_multi_layer_encoder()
+
+    def default_content_loss(
+        self, multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None
+    ) -> ops.FeatureReconstructionOperator:
+        if multi_layer_encoder is None:
+            multi_layer_encoder = self.default_multi_layer_encoder()
+        content_layer = "relu2_2"
+        content_encoder = multi_layer_encoder.extract_encoder(content_layer)
+        content_weight = 1e5
+        return ops.FeatureReconstructionOperator(content_encoder, score_weight=content_weight)
+
+    def default_style_loss(
+        self, multi_layer_encoder: Optional[enc.MultiLayerEncoder] = None
+    ) -> ops.MultiLayerEncodingOperator:
+
+        class GramOperator(ops.GramOperator):
```

Comment:
Should come from `pystiche`

Reply:
Can't. I'm going for compatibility here with the PyTorch example / paper implementation and they both do this. This is bug / oversight in their implementation.

Reply:
I've added an explanation through a comment.

Reply:
Make sense :)